server/src/main/java/org/elasticsearch/search/sort/SortValue.java

```diff
@@ -338,4 +346,69 @@ public Number numberValue() {
             return Double.NaN;
         }
     }
+
+    private static class EmptySortValue extends SortValue {
+
+        public static final String NAME = "empty";
+        private final byte dummyKey;
+
+        private EmptySortValue() {
+            this.dummyKey = (byte) 0;
+        }
+
+        @Override
+        public String getWriteableName() {
+            return NAME;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeByte(dummyKey);
+        }
+
+        @Override
+        public Object getKey() {
+            return dummyKey;
+        }
+
+        @Override
+        public String format(DocValueFormat format) {
+            return format.format(dummyKey).toString();
+        }
+
+        @Override
+        protected XContentBuilder rawToXContent(XContentBuilder builder) throws IOException {
+            return builder.value(dummyKey);
+        }
+
+        @Override
+        protected int compareToSameType(SortValue obj) {
+            EmptySortValue other = (EmptySortValue) obj;
```

Comment:
Maybe I can just return 0 here and in hashCode/equals.

Reply:
Yeah - that's probably fine.